pkg/standalone-utils/contracts/ProtocolFeeSplitter.sol

```diff
@@ -0,0 +1,130 @@
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/standalone-utils/IProtocolFeeSplitter.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/Authentication.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/helpers/VaultHelpers.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IProtocolFeesCollector.sol";
+
+interface Pool {
+    function getOwner() external returns (address);
+}
+
+/**
+ * @dev This contract is responsible for splitting the BPT profits collected
+ * by ProtocolFeeCollector between the pool's owner and Balancers DAO treasury
+ * Nothing happens for non-BPT tokens (WETH, WBTC, etc...)
+ */
+contract ProtocolFeeSplitter is IProtocolFeeSplitter, Authentication {
+    using Math for uint256;
+
+    // All fee percentages are 18-decimal fixed point numbers.
+    // Absolute maximum fee percentages (1e18 = 100%, 1e16 = 1%).
+    uint256 private constant _MAX_REVENUE_SHARING_FEE_PERCENTAGE = 50e16; // 50%
+    address private constant _DELEGATE_OWNER = 0xBA1BA1ba1BA1bA1bA1Ba1BA1ba1BA1bA1ba1ba1B;
+
+    IProtocolFeesCollector public immutable protocolFeesCollector;
+
+    // Balancer DAO Multisig
+    address public immutable treasury;
+
+    // Allows for a pool revenue override
+    mapping(bytes32 => uint256) public revenueSharePerPool;
+
+    constructor(IProtocolFeesCollector _protocolFeesCollector, address _treasury)
+        // The ProtocolFeeSplitter is a singleton, so it simply uses its own address to disambiguate action
+        // identifiers.
+        Authentication(bytes32(uint256(address(this))))
+    {
+        protocolFeesCollector = _protocolFeesCollector;
+        treasury = _treasury;
+    }
+
+    function setRevenueSharingFeePercentage(bytes32 poolId, uint256 newSwapFeePercentage)
+        external
+        override
+        authenticate
+    {
+        _require(newSwapFeePercentage <= _MAX_REVENUE_SHARING_FEE_PERCENTAGE, Errors.SPLITTER_FEE_PERCENTAGE_TOO_HIGH);
+        revenueSharePerPool[poolId] = newSwapFeePercentage;
+        emit PoolRevenueShareChanged(poolId, newSwapFeePercentage);
+    }
+
+    function collectFees(bytes32 poolId) external override {
+        address pool = VaultHelpers.toPoolAddress(poolId);
+        IERC20 bpt = IERC20(pool);
+
+        IProtocolFeesCollector protocolFeesCollectorMemory = protocolFeesCollector;
+        uint256 feeCollectorBptBalance = bpt.balanceOf(address(protocolFeesCollectorMemory));
+        if (feeCollectorBptBalance == 0) {
+            _revert(Errors.NO_BPT_FEES_COLLECTED);
+        }
+
+        // withdrawCollectedFees requires array of tokens, and we are only transfering BPT token
+        IERC20[] memory tokens = new IERC20[](1);
```

Comment:
I think you can do it like this:
IERC20[1] memory tokens = [bpt];

Reply:
https://www.tutorialspoint.com/solidity/solidity_arrays.htm

Reply:
Maybe it won't be accepted because this array is not dynamic though

Reply:
If this thing will work we can do it for all other array initializations

Reply:
Doesn't work, dynamic array